scipy/sparse/csgraph/_matching.pyx

```diff
@@ -297,8 +297,8 @@ def min_weight_full_bipartite_matching(biadjacency, maximize=False):
 
     Parameters
     ----------
-    biadjacency : sparse matrix
-        Biadjacency matrix of the bipartite graph: A sparse matrix in CSR, CSC,
+    biadjacency : sparse array or matrix
```

Comment:
Out of interest what determines which functions are changed to array or matrix Vs just array eg. `MaximumFlow`

Reply:
Good question!  I was not consistent throughout -- and it should be consistent.

Anywhere sparse array works, sparse matrix works too. 
The param description type now consistently (I hope) uses `sparse array or matrix`.

And I updated the np.array/np.matrix in some places to make those consistent... using `array_like` which was the predominant phrase. `array_like` is more accurate too. The input does not need to be ndarray or np.matrix. It can be subclasses too.

Thanks!!